### PR TITLE
KYLIN-4178: job scheduler safe mode

### PR DESCRIPTION
**Pull request by zhoukangcn**
See: https://issues.apache.org/jira/browse/KYLIN-4178

### CODE REVIEW

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java [hunk 1]
```diff
@@ -836,6 +836,14 @@ public Integer getSchedulerPollIntervalSecond() {
         return Integer.parseInt(getOptional("kylin.job.scheduler.poll-interval-second", "30"));
     }
 
+    public boolean isSchedulerSafeMode() {
+        return Boolean.parseBoolean(getOptional("kylin.job.scheduler.safemode", "false"));
+    }
+
+    public String getSchedulerSafeModeExecutableUUID() {
```
Comment (lshmouse) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: I think It's bettter to have a whitelist of project other than uuid. The job uuid may be changed.

Comment (zhoukangcn) [reply core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: change whitelist to project name

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java [hunk 2]
```diff
@@ -836,6 +837,14 @@ public Integer getSchedulerPollIntervalSecond() {
         return Integer.parseInt(getOptional("kylin.job.scheduler.poll-interval-second", "30"));
     }
 
+    public boolean isSchedulerSafeMode() {
+        return Boolean.parseBoolean(getOptional("kylin.job.scheduler.safemode", "false"));
+    }
+
+    public List<String> getSafeModeRunableProjects() {
+        return Arrays.asList(getOptionalStringArray("kylin.job.scheduler.safemode.runable-projects", null));
```
Comment (nichunen) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: Typo of "runable" here, typos in config items should not exist

Comment (zhoukangcn) [reply core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: fixed. Please help to review again.

**Comment by nichunen** [comment]
@zhoukangcn Hi，it's merged. Would you please submit another pr for the new configs of this enhancement in the branch "document"?

**Comment by zhoukangcn** [comment]
OK，I will update document  later.

**Pull request merged**